### PR TITLE
Trivial fixes for PHPCS errors and warnings

### PR DESCRIPTION
**Pull request by kdebisschop**
Just cleans up things pointed out by PHPCS.

### CODE REVIEW

File: docs/Extractors/Aggregator.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 Merge rows from a list of partial data iterators with a matching index.
 
 ```php
+use \Wizaplace\Etl\Etl;
```
Comment (kdebisschop) [inline docs/Extractors/Aggregator.md:6]: Adding this allows PHPStorm to do syntax checks, which is what revealed the typo in "transform" below.

**Comment by ecourtial** [comment]
We are going to merge your other PRs before rebasing this one.

File: src/Extractors/Csv.php [hunk 2]
```diff
@@ -118,7 +118,8 @@ protected function makeRow(array $row, array $columns): array
         foreach ($columns as $column => $index) {
             // The is bool is redundant, it is necessary because of PHP Stan, since we check the type in initialize()
             if ((bool) $this->throwError && false === array_key_exists($index - 1, $row)) {
-                throw new InvalidInputException("Row with index #{$this->currentRow} does not have the '{$column}' field.");
+                $message = "Row with index #{$this->currentRow} does not have the '{$column}' field.";
+                throw new InvalidInputException($message);
```
Comment (Nicolas-Masson-Wizaplace) [inline src/Extractors/Csv.php:122]: Would this be ok with phpcs ?
```suggestion
                throw new InvalidInputException(
                    "Row with index #{$this->currentRow} does not have the '{$column}' field."
                );
```

Comment (kdebisschop) [reply src/Extractors/Csv.php:122]: It is OK with PHPCS PSR scan, but it fails the PHP CS Fixer spec that you have included .php_cs.dist

**Comment by ecourtial** [comment]
@kdebisschop I merged the your PRs that were ready and approved.
I think you should rebase this one.

**Comment by kdebisschop** [comment]
Rebased.

Question for you ... since you are operating completely decoupled from the original work, does it make sense for me to refactor the tests so their method names also comply with PSR-12 ? That would allow more complete testing and also make it practical use a git pre-commit help for those of us inclined to use them...

(I have a pre-commit hook now that could be added as a sample, but it is made unduly complex by the fact that the tests do not comply with PSR-2)

**Comment by kdebisschop** [comment]
Hmmm... 

To answer my own question, there is raft of work required to get phpstan to accept the tests also, so it probably needs to be its own PR.

Also found that phpcs and php-cs-fixer disagree on whether anonymous classes should have a space before parens. For phpcs, it must be:

`$this->fakeLoader = new class () {`

and for php-cs-fixer it must be:

`$this->fakeLoader = new class () {`

The only solution I have found is to strictly follow the example at https://www.php-fig.org/psr/psr-12/#8-anonymous-classes and do:

`$this->fakeLoader = new class {`

While also setting `'new_with_braces' => false` in .php_cs.dist

**Comment by kdebisschop** [comment]
I also had to roll back a change to Xml.php that phpstan had required -- because now phpunit is failing on it. Phpunit seems more important, so I added the legacy exclusion in phpstan.neon instead.

And, composer update is now unhappy because the more recent versions of ocramius/package-versions require PHP 7.4. - that might need to be addressed in the composer.json file but I did not do that.

File: phpstan.neon [hunk 3]
```diff
@@ -25,3 +25,6 @@ parameters:
         -
             message: '~^Variable property access on ~'
             path: src/Step.php
+        -
+            message: '~^Dynamic call to static method XMLReader::open\(\)\.~'
+            path: src/Extractors\Xml.php
```
Comment (kdebisschop) [inline phpstan.neon:30]: PHPunit chokes on the static form so I had to roll back the change in Xml.php and add this exclusion.

File: src/Extractors/Xml.php [hunk 4]
```diff
@@ -70,10 +70,9 @@ public function extract(): \Generator
             }
         }
 
-        $this->reader = \XMLReader::open($this->input);
-        if (false === $this->reader) {
-            throw new XmlException('Could not open XMLReader.');
-        }
```
Comment (kdebisschop) [inline src/Extractors/Xml.php:76]: PHPUnit no longer likes the static version, even though both should be acceptable.

**Comment by kdebisschop** [comment]
There was no reason the test for FilePathTrait needed to use an anonymous class. I factored it out into a separate file so the conflict between phpcs and php-cs-fixer is not surfaced in the code we scan.

File: src/Extractors/Xml.php (hunk 4 above)
Comment (ecourtial) [reply src/Extractors/Xml.php:76]: That's weird. On my local machine and on the master branch build there is no error.

**Comment by ecourtial** [comment]
At the exception of my question regarding the XML call, everything is fine for me.

Regarding your global question about the future of this library. Yes, it is totally disconnected now from the original project which seems to be quite dead (many of my questions were never answered). I am considering a 2.0 version which would drop the support of PHP < 7.4, enforcing strict type everywhere. We also have other things to do, for instance we have an obsolete version of PHPUnit (8.x).

**Comment by kdebisschop** [comment]
I am at a loss to explain what causes PHPUnit to pass in some circumstances and fail in others. Nonetheless, this is what I see locally if I use the static version -- failures actually in both phpunit and phpstan:

```
Karls-MBP-2:php-etl > vendor/bin/phpunit tests
PHPUnit 8.5.8 by Sebastian Bergmann and contributors.

...........................................................EE..  63 / 138 ( 45%)
............................................................... 126 / 138 ( 91%)
............                                                    138 / 138 (100%)

Time: 258 ms, Memory: 24.00 MB

There were 2 errors:

1) Tests\Extractors\XmlTest::custom_loop_path
Non-static method XMLReader::open() should not be called statically

/Users/debisschop/PhpstormProjects/php-etl/src/Extractors/Xml.php:73
/Users/debisschop/PhpstormProjects/php-etl/tests/Extractors/XmlTest.php:33

2) Tests\Extractors\XmlTest::custom_fields_within_the_loop_path
Non-static method XMLReader::open() should not be called statically

/Users/debisschop/PhpstormProjects/php-etl/src/Extractors/Xml.php:73
/Users/debisschop/PhpstormProjects/php-etl/tests/Extractors/XmlTest.php:56

ERRORS!
Tests: 138, Assertions: 272, Errors: 2.
Karls-MBP-2:php-etl > vendor/bin/phpstan analyze --memory-limit=512M src
Note: Using configuration file /Users/debisschop/PhpstormProjects/php-etl/phpstan.neon.
 45/45 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ------------------------------------------------------------------------------------------------------- 
  Line   Extractors/Xml.php                                                                                     
 ------ ------------------------------------------------------------------------------------------------------- 
         Ignored error pattern ~^Dynamic call to static method XMLReader::open\(\)\.~ in path                   
         /Users/debisschop/PhpstormProjects/php-etl/src/Extractors/Xml.php was not matched in reported errors.  
 ------ ------------------------------------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 1 error          
```                                                                                        
 
In fact, the PHP web site suggests that it may be called statically:

> ### Return Values
>
> Returns TRUE on success or FALSE on failure. If called statically, returns an XMLReader or FALSE on failure.

but the next part is informative:

> ### Errors/Exceptions
>
> This method may be called statically, but will issue an E_STRICT error.

For that reason, I believe the original dynamic call is preferable, even though it requires adding an exception to phpstan.neon. But I'm certainly open to counter-arguments.

**Review by ecourtial — APPROVED** [review]

**Review by Nicolas-Masson-Wizaplace — APPROVED** [review]

**Pull request merged**